lstests/doc/README.md

```diff
@@ -0,0 +1,688 @@
+
+```
+thread-1: first half
+thread-1: second half
+thread-2: first half
+thread-2: second half
+```
+
+or
+
+```
+thread-2: first half
+thread-2: second half
+thread-1: first half
+thread-1: second half
+```
+
+but will never be of the following form
+
+```
+thread-2: first half
+thread-1: first half
+thread-1: second half
+thread-2: second half
+```
+
+#### Log Levels
+
+| level   | value |
+| ------- | ----- |
+| `OFF`   | 100   |
+| `FLAG`  | 90    |
+| `FATAL` | 60    |
+| `ERROR` | 50    |
+| `WARN`  | 40    |
+| `INFO`  | 30    |
+| `DEBUG` | 20    |
+| `TRACE` | 10    |
+| `ALL`   | 0     |
+
+The `LogLevel` class accepts either a numeric value or a case-insensitive string for the level. While there are text names given for the level values, all the integers in `[0, 100]` are valid log levels.
+
+```python
+clus.log("Some warning information.", level="warn")
+```
+
+is equivalent to
+
+```python
+clus.log("Some warning information.", level=40)
+```
+
+or
+
+```python
+clus.warn("Some warning information.")
+```
+
+As another example,
+
+```python
+clus.log("Some info more prominent than INFO but not yet a WARN", level=35)
+```
+
+This line will be written to any logging destination with a threshold at or lower than 35. It will be printed out on the screen if the `ScreenWriter` has an `INFO`-level threshold. It will be written to a log file if there is a file writer `FileWriter(threshold=35)`. But it will not appear on the log file that captures only `WARN`-or-higher-level information.
+
+### `Service`
+
+A `Service` object represents the connection with the launcher service running in the background. Once a `Service` object is created, it will try to connect the launcher service automatically.
+
+After configuration, the `Service` will change the working directory, register (and possibly *modify*) the Logger, and then connect to launcher service. *Currently, it is only possible to connect to a local launcher service.*
+
+If there is already an existing launcher service running in the background, the `Service` object will by default connect to it (without starting a ew one). It is possible to override this default behavior by requesting to always start a new launcher service and/or to kill all the existing launcher service(s).
+
+A detailed explanation of the parameters to initialize a `Service` object can be found in its docstring.
+
+     Parameters
+     ----------
+    logger : Logger
+        Logger object which controls logging behavior.
+    addr : str
+        IP address of launcher service.
+        Currently, only local launcher service is supported. That is, only
+        default value "127.0.0.1" is supported.
+    port : int
+        Listening port of launcher service.
+        If there are multiple launcher service running in the background,
+        they must have different listening ports.
+        Default is 1234.
+    wdir : str
+        Working directory.
+        Default is the build folder.
```

Comment:
Am I correct in understanding that the default working directory assumes a `build` folder within the source directory? I think the default should be the current working directory. The CMake already changes the working directory to the build folder before executing the lstests scripts so it should still work well with that. Furthermore making it the current working directory rather then the `build` folder is more flexible since that default would support people who want to explicitly run certain tests from out-of-source builds and do not use the build scripts (i.e. most of the core developers of EOSIO).

Reply:
Thanks for the comment. I take this point and will work to make the current working directory the default value for `wdir`.